weather_mv/loader_pipeline/netcdf_loader_test.py

```diff
@@ -203,5 +211,102 @@ def test_extract_rows_nan(self):
         self.assertRowsEqual(actual, expected)
 
 
+def _handle_missing_grib_be(f):
+    @wraps(f)
+    def decorated(*args, **kwargs):
+        try:
+            return f(*args, **kwargs)
+        except ValueError as e:
+            # Some setups may not have Cfgrib installed properly. Ignore tests for these cases.
+            e_str = str(e)
+            if "Consider explicitly selecting one of the installed engines" not in e_str or "cfgrib" in e_str:
+                raise
+
+    return decorated
+
+
+class ExtractRowsGribSupportTest(ExtractRowsTestBase):
+
+    def setUp(self) -> None:
+        super().setUp()
+        self.tmp_dir = tempfile.TemporaryDirectory()
+        self.test_data_path = f'{self.test_data_folder}/test_data_grib_single_timestep'
+
+    def tearDown(self):
+        # Close the file, the directory will be removed after the test
+        self.tmp_dir.cleanup()
+
+    @_handle_missing_grib_be
+    def test_extract_rows(self):
+        actual = next(extract_rows(self.test_data_path, tmp_dir=self.tmp_dir.name))
+        expected = {
+            'data_import_time': '1970-01-01T00:00:00+00:00',
+            'data_first_step': '2021-10-18T06:00:00+00:00',
+            'data_uri': self.test_data_path,
+            'latitude': 90.0,
+            'longitude': -180.0,
+            'number': 0,
+            'step': 0.0,
+            'surface': 0.0,
+            'time': '2021-10-18T06:00:00+00:00',
+            'valid_time': '2021-10-18T06:00:00+00:00',
+            'z': 1.42578125,
+        }
+        self.assertRowsEqual(actual, expected)
+
+    @_handle_missing_grib_be
+    def test_multiple_editions(self):
+        self.test_data_path = f'{self.test_data_folder}/test_data_grib_multiple_edition_single_timestep.bz2'
```

Comment:
In the future, we should write a script to generate these testdata files. Checking them in slows down how fast we can clone the repo.